utils/findLocaleStrings.js

```diff
@@ -0,0 +1,115 @@
+import {transform} from 'babel-core';
+import traverse from 'babel-traverse';
+import fs from 'fs';
+import glob from 'glob';
+import path from 'path';
+
+const pathToSrc = path.join(__dirname, '../src');
+const srcGlob = path.join(pathToSrc, '**/*.js');
+
+let EXIT_CODE = 0;
+const localizeRE = /(^|[\.])(_|localize)$/;
+
+findLocaleStrings();
+
+function findLocaleStrings() {
+  glob(srcGlob, (err, files) => {
+    if (err) {
+      EXIT_CODE = 1;
+      console.log(err);
+      return;
+    }
+
+    const dict = {};
+    let hasTranslation = false;
+    let maxLen = 0;
+
+    files.forEach(file => {
+      const code = fs.readFileSync(file, 'utf-8');
+      const filePartialPath = file.substr(pathToSrc.length);
+      const ast = transform(code, {
+        presets: ['react', 'es2015', 'stage-2'],
+      }).ast;
+
+      traverse(ast, {
+        enter(path) {
+          if (
+            path.node.type === 'CallExpression' &&
+            path.node.callee.name === '_'
+          ) {
+            const strNode = path.node.arguments[0];
+            let strNodeValue = strNode.value;
+
+            if (path.node.arguments.length !== 1) {
+              logError(file, path.node, 'Localize takes 1 args');
+            }
+
+            if (
+              ['StringLiteral', 'BinaryExpression'].indexOf(strNode.type) < 0
+            ) {
+              logError(
+                file,
+                path.node,
+                `The localization function takes a string as argument, instead it received a ${
+                  strNode.type
+                }`
+              );
+            }
+
+            if (strNode.type === 'BinaryExpression') {
+              strNodeValue = path.get('arguments')[0].evaluate().value;
+              if (typeof strNodeValue !== 'string') {
+                logError(
+                  file,
+                  path.node,
+                  `The localization function takes a string as argument, instead it received a ${typeof strNodeValue}`
+                );
+              }
+            }
+
+            if (!dict[strNodeValue]) {
+              dict[strNodeValue] =
+                filePartialPath + ':' + strNode.loc.start.line;
+              maxLen = Math.max(maxLen, strNodeValue.length);
```

Comment:
might want to put a limit on this... 80? so a few of the longest strings would not have the line-number comment lined up with the rest (just give them at least 2 spaces), but the majority of them would and it'd be closer and easier to use.

Reply:
I kind of like that they're all aligned, even if there are some long stings, I find it's more consistent for my eyes :) and makes me notice the super long strings, i.e. may be a sign that they'd need to be shortened...